### ISSUE TITLE
Does the package stil work on ST-4189?

### ISSUE THREAD

**Issue by nixtoshi** [jfcherng-sublime/ST-RainbowIndent#8]
I recently installed the package on Sublime Text 4189 but I can't get it to work.

I pasted the color scheme into color scheme customization file inside the `"rules": [] brackets` and I set the package's settings to:

Package settings file:
```
{
// Scopes in the color scheme to be used for indent levels.
    "level_colors": [
        "region.indent.0",
        "region.indent.1",
        "region.indent.2",
        "region.indent.3",
        "region.indent.4",
        "region.indent.5",
    ],
    "level_style": "line",
    "enabled_selector": "",
    // The max file size to enable the plugin. Set to negative value to disable the limit.
    "file_size_limit": 10485760, // about 10 MB
}
```

Yet I don't see the colored indentation. I'm using the GitHub Dimmed theme.

Color scheme customization file:
```
// Documentation at https://www.sublimetext.com/docs/color_schemes.html
{
  "variables":
  {
  },
  "globals":
  {
  },
  "rules":
  [
    {
      // red
      "foreground": "rgba(229, 57, 53, 0.35)",
      "background": "rgba(229, 57, 53, 0.2)",
      "scope": "region.indent.0"
    },
    {
      // green
      "foreground": "rgba(67, 160, 71, 0.35)",
      "background": "rgba(67, 160, 71, 0.2)",
      "scope": "region.indent.1"
    },
    {
      // blue
      "foreground": "rgba(30, 136, 229, 0.35)",
      "background": "rgba(30, 136, 229, 0.2)",
      "scope": "region.indent.2"
    },
    {
      // orange
      "foreground": "rgba(251, 140, 0, 0.35)",
      "background": "rgba(251, 140, 0.2)",
      "scope": "region.indent.3"
    },
    {
      // purple
      "foreground": "rgba(142, 36, 170, 0.35)",
      "background": "rgba(142, 36, 170, 0.2)",
      "scope": "region.indent.4"
    },
    {
      // cyan
      "foreground": "rgba(0, 172, 193, 0.35)",
      "background": "rgba(0, 172, 193, 0.2)",
      "scope": "region.indent.5"
    }
  ]
}
```

I'm using spaces as indents instead of tabs. I wanted to have the indents colored in HTML, CSS and JS files.
I tried switching between tabs and spaces for indentation and still, no color.

**Comment by jfcherng** [comment]
I am the main user of this plugin and I always use the latest dev build (4190 at this moment) in my daily work. So I most likely will be the first one who reports/fixes this if it doesn't work.

**Comment by jfcherng** [comment]
This plugin works "without" any customization (no settings. no color rules). You may 

- Make sure the plugin is enabled. Package Control will disable a plugin during updating and failed to enable it back sometimes.
- Test it without any customization first.
- Any error message in ST console?
- Is the indent style in the window's bottom-right corner matches your code?
- Reproducible source code example? Not working for ANY code or just the code you said in this issue?

**Comment by nixtoshi** [comment]
I got it working by deleting ST's folder on `~/Library/Application Support/` and doing a fresh install. I noticed Sublime Text 4 was using a folder called `Sublime Text 3` on `Application Support` instead of just `Sublime Text` after Sublime Text auto-updated, maybe this was part of the issue.

When I did a fresh install and Package Control got updated to version `4.0` everything worked again, the package is working alongside my customized Github theme and custom color scheme as well. I think the issue might have been that I was using the latest version of Package Control for ST3 instead of the latest version for ST4, and it wasn't properly downloading the python dependencies to run new and older packages. Package control also showed that everything was up to date including itself.

After some reading online, Package Control 4.0 only works on ST4, so I was probably using an older Package Control that wasn't aware I had updated to ST4.

These were the error logs when RainbowIndent wasn't working (I later realized some other packages weren't working either):

```
error

startup, version: 4189 osx arm64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/sublime_text
application: /Applications/Sublime Text.app
working dir: /
packages path: /Users/dalovar/Library/Application Support/Sublime Text 3/Packages
state path: /Users/dalovar/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.276204
OpenGL Context Information:
  GL API Version: 4.1 Metal - 88
  GLSL Version: 4.10
  Vendor: Apple
  Renderer: Apple M2 Max
startup time: 0.388843
environment variables loaded using: /bin/zsh -l
first paint time: 0.406063
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors
reloading python 3.3 plugin ayu.activation
reloading plugin Default.comment
reloading python 3.3 plugin ayu.Icons
reloading python 3.3 plugin ReverseCharacters.reverse_characters
reloading python 3.3 plugin SortBy.sortby
reloading plugin Default.convert_color_scheme
reloading python 3.3 plugin SortBy.sortbyregex
reloading python 3.3 plugin HTML-CSS-JS Prettify.HTMLPrettify
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.html_print
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.keymap
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin Package Control.plugin
reloading plugin RainbowIndent.boot
Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/Lib/python38/sublime_plugin.py", line 325, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.8/importlib/__init__.py", line 127, in import_module
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 868, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/boot.py", line 15, in <module>
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/plugin/__init__.py", line 3, in <module>
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/plugin/commands/__init__.py", line 3, in <module>
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/plugin/commands/rainbow_indent_view_toggle.py", line 7, in <module>
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/plugin/listener.py", line 6, in <module>
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/plugin/helpers.py", line 6, in <module>
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/plugin/settings.py", line 8, in <module>
  File "/Users/dalovar/Library/Application Support/Sublime Text 3/Installed Packages/RainbowIndent.sublime-package/plugin/data_types.py", line 11, in <module>
ModuleNotFoundError: No module named 'typing_extensions'
plugins loaded
Package Control: Installing 2 libraries...
Package Control: The library "more-itertools" is not available for Python 3.8
Package Control: The library "typing-extensions" is not available for Python 3.8
Package Control: Skipping automatic upgrade, last run at 2024-12-26 22:47:30, next run at 2024-12-27 06:47:30 or after
```

And these were the logs when it did work:



```
startup, version: 4189 osx arm64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/sublime_text
application: /Applications/Sublime Text.app
working dir: /
packages path: /Users/dalovar/Library/Application Support/Sublime Text/Packages
state path: /Users/dalovar/Library/Application Support/Sublime Text/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/dalovar/Library/Application Support/Sublime Text/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.220303
OpenGL Context Information:
  GL API Version: 4.1 Metal - 88
  GLSL Version: 4.10
  Vendor: Apple
  Renderer: Apple M2 Max
startup time: 0.278877
environment variables loaded using: /bin/zsh -l

first paint time: 0.292333
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors

reloading plugin Default.comment



reloading plugin Default.convert_color_scheme


reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.html_print
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.keymap
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin Compare Side-By-Side.sbs_compare
reloading plugin Package Control.plugin
plugins loaded
Package Control: Skipping automatic upgrade, last run at 2024-12-26 23:10:48, next run at 2024-12-27 07:10:48 or after
Package Control: Searching updates...
Package Control: All specified packages up-to-date!
Package Control: pausing indexer
ignored packages updated to: ["RainbowIndent", "Vintage"]
reloading settings Packages/User/Package Control.sublime-settings
reloading settings Packages/User/Preferences.sublime-settings
Package Control: Installed library "more-itertools" 10.5.0 for Python 3.8
Package Control: Installed library "typing-extensions" 4.12.2 for Python 3.8
Package Control: Installed package "RainbowIndent" 0.1.6
ignored packages updated to: ["Vintage"]
reloading settings Packages/User/Package Control.sublime-settings
reloading plugin RainbowIndent.boot
reloading settings Packages/User/Preferences.sublime-settings
Package Control: resuming indexer
reloading settings Packages/User/Preferences.sublime-settings
```
